IdentityCore/src/cache/token/MSIDAccountCacheItem.m

```diff
@@ -22,95 +22,143 @@
+    result &= (!self.name || !item.name) || [self.name isEqualToString:item.name];
+    result &= (!self.realm || !item.realm) || [self.realm isEqualToString:item.realm];
+    result &= (!self.clientInfo || !item.clientInfo) || [self.clientInfo.rawClientInfo isEqualToString:item.clientInfo.rawClientInfo];
+    result &= (!self.environment || !item.environment) || [self.environment isEqualToString:item.environment];
+    result &= (!self.alternativeAccountId || !item.alternativeAccountId) || [self.alternativeAccountId isEqualToString:item.alternativeAccountId];
+    return result;
+}
+
+#pragma mark - NSObject
+
+- (NSUInteger)hash
+{
+    NSUInteger hash = [super hash];
+    hash = hash * 31 + self.accountType;
+    hash = hash * 31 + self.uniqueUserId.hash;
+    hash = hash * 31 + self.legacyUserId.hash;
+    hash = hash * 31 + self.username.hash;
+    hash = hash * 31 + self.givenName.hash;
+    hash = hash * 31 + self.middleName.hash;
+    hash = hash * 31 + self.familyName.hash;
+    hash = hash * 31 + self.name.hash;
+    hash = hash * 31 + self.realm.hash;
+    hash = hash * 31 + self.clientInfo.hash;
+    hash = hash * 31 + self.environment.hash;
+    hash = hash * 31 + self.alternativeAccountId.hash;
+    return hash;
+}
+
+#pragma mark - NSCopying
+
+- (id)copyWithZone:(NSZone *)zone
+{
+    MSIDAccountCacheItem *item = [[self class] allocWithZone:zone];
+    item.accountType = self.accountType;
+    item.uniqueUserId = [self.uniqueUserId copyWithZone:zone];
+    item.legacyUserId = [self.legacyUserId copyWithZone:zone];
+    item.username = [self.username copyWithZone:zone];
+    item.givenName = [self.givenName copyWithZone:zone];
+    item.middleName = [self.middleName copyWithZone:zone];
+    item.familyName = [self.familyName copyWithZone:zone];
+    item.name = [self.name copyWithZone:zone];
+    item.realm = [self.realm copyWithZone:zone];
+    item.clientInfo = [self.clientInfo copyWithZone:zone];
+    item.environment = [self.environment copyWithZone:zone];
+    item.alternativeAccountId = [self.alternativeAccountId copyWithZone:zone];
+    return item;
 }
 
 #pragma mark - JSON
 
 - (instancetype)initWithJSONDictionary:(NSDictionary *)json error:(NSError **)error
 {
-    if (!(self = [super initWithJSONDictionary:json error:error]))
+    if (!(self = [super init]))
     {
         return nil;
     }
-    
-    // Authority account ID
-    _legacyUserIdentifier = json[MSID_ACCOUNT_ID_CACHE_KEY];
-    
-    /* Optional fields */
-    // First name
-    _firstName = json[MSID_FIRST_NAME_CACHE_KEY];
-    
-    // Last name
-    _lastName = json[MSID_LAST_NAME_CACHE_KEY];
-    
-    // Account type
+
+    _json = json;
+
     _accountType = [MSIDAccountTypeHelpers accountTypeFromString:json[MSID_AUTHORITY_TYPE_CACHE_KEY]];
-    
-    // Extensibility
-    _additionalAccountFields = json;
-    
+    _legacyUserId = json[MSID_ACCOUNT_ID_CACHE_KEY];
```

Comment:
General Q not particularly specific to this.
If some constants are only used in one class, do they really need to be classified constants outside?